src/frontend/src/scheduler/local.rs

```diff
@@ -129,9 +112,20 @@ impl LocalQueryExecution {
     }
 
     pub fn stream_rows(self) -> LocalQueryStream {
-        LocalQueryStream {
-            data_stream: self.run(),
-        }
+        let (sender, receiver) = mpsc::channel(10);
+        let mut data_stream = self.run().map(|r| r.map_err(|e| Box::new(e) as BoxedError));
+
+        spawn_blocking(move || {
+            block_on(async {
+                while let Some(r) = data_stream.next().await {
+                    if (sender.send(r).await).is_err() {
+                        tracing::info!("Receiver closed.");
+                    }
+                }
+            })
+        });
```

Comment:
Blocking operations are not supported in simulation. You may change it to:
```suggestion
        let future = async {
            while let Some(r) = data_stream.next().await {
                if (sender.send(r).await).is_err() {
                    tracing::info!("Receiver closed.");
                }
            }
        };
        #[cfg(madsim)]
        spawn(future);
        #[cfg(not(madsim))]
        spawn_blocking(move || block_on(future));
```